data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/GreaterThanOperator.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.expression;
+
+import org.opensearch.dataprepper.expression.antlr.DataPrepperExpressionParser;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiFunction;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+class GreaterThanOperator implements Operator<Boolean> {
+    private static final Map<String, BiFunction<Object, Object, Boolean>> GREATER_THAN_ON =
+            new HashMap<String, BiFunction<Object, Object, Boolean>>() {{
+                put(Integer.class.getName() + "_" + Integer.class.getName(), (a, b) -> (Integer) a > (Integer) b);
+                put(Integer.class.getName() + "_" + Float.class.getName(), (a, b) -> (Integer) a > (Float) b);
+                put(Float.class.getName() + "_" + Integer.class.getName(), (a, b) -> (Float) a > (Integer) b);
+                put(Float.class.getName() + "_" + Float.class.getName(), (a, b) -> (Float) a > (Float) b);}};
+
+    @Override
+    public Integer getSymbol() {
+        return DataPrepperExpressionParser.GT;
+    }
+
+    @Override
+    public Boolean evaluate(Object... args) {
+        checkArgument(args.length == 2, "Operands length needs to be 2.");
```

Comment:
I do not have the full context. But my guess is that L31 should always given the expression parser and correct parseTreeEvaluator evaluation logic that is supposed to be consistent with the ANTLR grammar

Reply:
There are a couple of safety checks before the operator is called.

1. The Grammar should enforce that operands are always provided the correct number of arguments
2. The Operator provides a required number of arguments that will be pulled from the arg stack before evaluate is called.

The exception message could be more specific, saying something like "Greater than operator requires two arguments". The evaluator class will then catch that exception and can provide more context on where in the statement the error occurred. I don't think it would make sense to expose that level of information to the operator instance.

Issue [1003](https://github.com/opensearch-project/data-prepper/issues/1003) has a sequence flow diagram to give additional context.

Reply:
@sbayer55 Thanks. IMO, to provide more specific exception message with operator type encoded, we might need to rethink on the design of NumericCompareOperator or NegateOperator that have been proposed. @cmanning09 thoughts?

Reply:
Have you considered a negate operator that internally holds a reference to another operator whose results are negated?